src/applications/MachineShopSimulator.java

```diff
@@ -42,40 +42,55 @@ static boolean moveToNextMachine(Job theJob, SimulationResults simulationResults
         }
     }
 
+    private static int getMachine(Job theJob) {
+        return ((Task) theJob.getTaskQ().getFrontElement()).getMachine();
+    }
+
     /**
      * change the state of theMachine
      * 
      * @return last job run on this machine
      */
-    static Job changeState(int theMachine) {// Task on theMachine has finished,
-                                            // schedule next one.
-        Job lastJob;
-        if (machine[theMachine].getActiveJob() == null) {// in idle or change-over
-                                                    // state
-            lastJob = null;
-            // wait over, ready for new job
-            if (machine[theMachine].getJobQ().isEmpty()) // no waiting job
+    static Job changeState(int theMachine) {// Task on theMachine has finished, schedule next one.
+        Machine currentMachine = machine[theMachine];
+        Job lastJob = null;
+        if (currentMachine.getActiveJob() == null) {
+            // in idle or change-over state wait over, ready for new job
+            if (currentMachine.getJobQ().isEmpty()) // no waiting job
                 eList.setFinishTime(theMachine, largeTime);
-            else {// take job off the queue and work on it
-                machine[theMachine].setActiveJob((Job) machine[theMachine].getJobQ()
-                        .remove());
-                machine[theMachine].setTotalWait(machine[theMachine].getTotalWait() + timeNow
-                        - machine[theMachine].getActiveJob().getArrivalTime());
-                machine[theMachine].setNumTasks(machine[theMachine].getNumTasks() + 1);
-                int t = machine[theMachine].getActiveJob().removeNextTask();
+            else {
+                // take job off the queue and work on it
+                getNextJob(currentMachine);
+                setReturnTime(currentMachine);

```

Comment:
This is a lot more clear and really resolves the chain.

Reply:
I agree. This code is a lot easier to read than it was before.